utils/checks.stanza

```diff
@@ -179,34 +194,33 @@ pcb-check no-connect-pins-not-connected (module:JITXObject) :
       fail-message =         "Pin %_ is marked no-connect and is incorrectly connected to other pins" % [ref(p)],
       locators =             [p])
 
-; A check used to verify the electrical properties of 
+; A check used to verify the electrical properties of
 ; digital i/o pins.
-pcb-check digital-io-levels (module:JITXObject) : 
-  val all-pins = all-pins(module)
-  val io-pins  = to-tuple $ filter(is-digital-io?, all-pins)
-  for p in io-pins do : 
-    val all-connected-pins = all-pins 
-      $> filter{{connected?([p, _0]) and _0 != p}, _}
-      $> to-tuple
-    val connected-io-pins = all-connected-pins
-      $> filter{is-digital-io?, _}
-      $> to-tuple
-    val connected-out-pins = all-connected-pins 
-      $> filter{has-property?{_.digital-output}, _}
-      $> to-tuple
+pcb-check digital-io-levels (module:JITXObject) :
+  val io-pins = filter(is-digital-io?, all-populated-pins(module))
+
+  ; FIXME! Could we split them first in connected groups and then run the loop only once per connected group?
+  ; Issue:: some of the logic is assymetrical and considers this p as special in the connected group considered
+  for p in io-pins do :
+    val all-connected-pins = populated-connected-pins(p)
 
+    ; FIXME!: not calling digital-output? that accepts digital-io property?
     if has-property?(p.digital-output) :
+      val connected-out-pins = all-connected-pins $> filter{has-property?{_.digital-output}, _}
+                                                  $> to-tuple
       #CHECK(condition =            empty?(connected-out-pins)
-            name =                 "IO pin checks"
-            description =          "Check io pin properties"
-            category =             "pin-check"
-            subcheck-description = "Check output pins are correctly connected",
-            pass-message =         "Pin %_ is marked as an output and not connected to any output pins" % [ref(p)],
-            fail-message =         "Pin %_ is marked as an output and connected to the following output pins: %," % [ref(p), seq(ref, connected-out-pins)],
-            locators =             [p])
+             name =                 "IO pin checks"
+             description =          "Check io pin properties"
+             category =             "pin-check"
+             subcheck-description = "Check output pins are correctly connected",
+             pass-message =         "Pin %_ is marked as an output and not connected to any output pins" % [ref(p)],
+             fail-message =         "Pin %_ is marked as an output and connected to the following output pins: %," % [ref(p), seq(ref, connected-out-pins)],
+             locators =             [p])
+
+    val connected-io-pins = filter(is-digital-io?, all-connected-pins)
```

Comment:
Rename `is-digital-io?` as `is-digital-pin?`.
For each connected group, only once we want to filter on io vs input vs output. If needed in some logic we can uses that input and output are subset of io